tests/rbd/test_rbd_compression.py

```diff
@@ -0,0 +1,222 @@
+"""Test case covered -
+    CEPH-83574644 - Validate "rbd_compression_hint" config
+    settings on globally, Pool level, and image level.
+
+    Pre-requisites :
+    1. Cluster must be up and running with capacity to create pool
+       (At least with 64 pgs)
+    2. We need atleast one client node with ceph-common package,
+       conf and keyring files
+
+    Test Case Flow:
+    1. Create a pool and an Image, write some data on it
+    2. set bluestore_compression_mode to passive to enable rbd_compression_hint feature
+    3. verify "rbd_compression_hint" to "compressible" on global, pool and image level
+    4. verify "rbd_compression_hint" to "incompressible" on global, pool and image level
+    5. Repeat the above steps for ecpool
+"""
+import json
+
+from tests.rbd.rbd_utils import initial_rbd_config
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def validate_compression(self, mode, image_spec, **kw):
+    """
+    Method to validate after setting rbd_compression_hint
+    data is getting compressed or not.
+
+    Args:
+        mode: type of compression hints (compressible or incompressible)
+        image_spec: poolname + imagename
+        **kw: test data
+
+    Returns:
+        int: The return value. 0 for success, 1 for failure.
+    """
+    config = kw.get("config")
+    io = config.get("io_total")
+    kw["io"] = io
+    kw["imagespec"] = image_spec
+    pool_name = image_spec.split("/")[0]
+
+    # running io to check compression of data
+    self.rbd_bench(**kw)
+
+    out = self.exec_cmd(cmd="ceph df detail --format json-pretty", output=True)
+    output = json.loads(out.rstrip())
+    pool_info = output["pools"]
+
+    # Iterate over the pools to get compressed data information
+    for pool in pool_info:
+        if pool["name"] == pool_name:
+            compressed_data = pool["stats"]["compress_bytes_used"]
+            break
+    log.debug(pool["stats"])
```

Comment:
I think this log.debug should be inside the for loop isn't it?

Reply:
@sunilangadi2 Also, please add more info to debug message, just printing numbers wouldn't help

Reply:
done added

Reply:
okay with log message